internal/middleware/authz.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2024 Canonical.
+
+package middleware
+
+import (
+	"net/http"
+
+	cofga "github.com/canonical/ofga"
+	"github.com/go-chi/chi/v5"
+	"github.com/juju/names/v5"
+
+	"github.com/canonical/jimm/v3/internal/jujuapi"
+	ofganames "github.com/canonical/jimm/v3/internal/openfga/names"
+)
+
+// AuthorizeUserForModelAccess extract the user from the context, and checks for permission on the model uuid extracted from the path.
+func AuthorizeUserForModelAccess(next http.Handler, jimm jujuapi.JIMM, accessNeeded cofga.Relation) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		user, err := IdentityFromContext(ctx)
+		if err != nil {
```

Comment:
Some space between blocks would be nice, it's frustrating to read when all bulked together... i.e.,:

```go
		modelUUID := chi.URLParam(r, "uuid")
		if modelUUID == "" {
			w.WriteHeader(http.StatusUnauthorized)
			_, _ = w.Write([]byte("cannot find uuid in URL path"))
			return
		}

		modelTag := names.NewModelTag(modelUUID)

		switch accessNeeded {
		case ofganames.ReaderRelation:
```